tests/test_class_attributes.rs

```diff
@@ -224,7 +224,7 @@ macro_rules! test_case {
                 let struct_obj = struct_class.call0().unwrap();
                 assert!(struct_obj.setattr($renamed_field_name, 2).is_ok());
                 let attr = struct_obj.getattr($renamed_field_name).unwrap();
-                assert_eq!(2, attr.extract().unwrap());
+                assert_eq!(2, attr.extract::<u8>().unwrap());
```

Comment:
So just completing my read of the full diff, this is interesting. 

What I see here is that the introduction of these implementations has changed type inference here. It's a potentially breaking change. 

Is it a problem? I don't _think_ so; user code could already define other types implementing `FromPyObject` and `PartialEq<u8>` and thus have the same inference failure.

And I believe that this new `PartialEq` implementation here still holds potential convenience and usability improvements.

Reply:
I see how this can be a breaking change. However, I don't think it's a major issue.

Ultimately, I'll leave the final decision to the maintainers haha :smiley: .

Reply:
Agreed, I think we move ahead with this.

Reply:
I don't consider inference failures like this as breaking changes, so I'm also fine with this.